web/libs/editor/src/mixins/HighlightMixin.js

```diff
@@ -29,30 +33,44 @@ export const HighlightMixin = types
         classNames.push(HIGHLIGHT_NO_LABEL_CN);
       }
 
+      if (self.selected) {
+        classNames.push(STATE_CLASS_MODS.active);
+      }
+
       // in this case labels presence can't be changed from settings — manual mode
       if (isDefined(self.parent.showlabels)) {
         classNames.push("htx-manual-label");
       }
 
       return classNames;
     },
-    get styles() {
-      const { className } = self;
-      const activeColorOpacity = 0.8;
-      const color = self.getLabelColor();
-      const initialActiveColor = Utils.Colors.rgbaChangeAlpha(color, activeColorOpacity);
-
+    /**
+     * Generate styles for region and active region, but with a lighter background is that's a resized region,
+     * so the selection of the same color will give the original color of active region.
+     * @see getColors
+     * @param {string} className
+     * @param {object} colors see `getColors()`
+     * @param {boolean} resize lighter background for resized region or original one for active region
+     * @returns {string} styles to apply to the region
+     */
+    generateStyles(className, colors, resize = false) {
       return `
         .${className} {
-          background-color: ${color} !important;
+          background-color: ${colors.background} !important;
           border: 1px dashed transparent;
         }
         .${className}.${STATE_CLASS_MODS.active}:not(.${STATE_CLASS_MODS.hidden}) {
-          color: ${Utils.Colors.contrastColor(initialActiveColor)} !important;
-          background-color: ${initialActiveColor} !important;
+          color: ${colors.activeText} !important;
+          background-color: ${resize ? colors.resizeBackground : colors.activeBackground} !important;
         }
       `;
     },
+    get styles() {
+      return this.generateStyles(self.className, self.getColors());
```

Comment:
It looks like styles and resizeStyles are the same thing and could simply be recalculated based on an internal volatile flag instead of creating generators and complicating dependencies.

Reply:
Unfortunately the whole styles approach here is too complex anyway. For example, styles are changed by the method and colors they rely on are not in `views` for some reason. And colors are multiple values. But I'll add this as an idea to follow up ticket. And I added a todo already about colors and views.